src/ui/public/notify/__tests__/notifier.js

```diff
@@ -43,8 +51,47 @@ describe('Notifier', function () {
       expect(notify('error').title).to.equal('Error');
     });
 
-    it('sets lifetime to Infinity', function () {
-      expect(notify('error').lifetime).to.equal(Infinity);
+    it('sets lifetime to 5 minutes', function () {
+      expect(notify('error').lifetime).to.equal(300000);
+    });
+
+    it('sets timeRemaining and deprecates', function () {
+      var notif = notify('error');
+
+      expect(notif.timeRemaining).to.equal(300);
+      $interval.flush(1000);
+      expect(notif.timeRemaining).to.equal(299);
+    });
+
+    it('closes notification on lifetime expiry', function () {
+      var expectation = sinon.mock();
+      var notif = notifier.error(message, expectation);
+
+      expectation.once();
+      expectation.withExactArgs('ignore');
```

Comment:
:clap: For assertions via mocks. I don't know why we (myself included) don't do that more often.